### ISSUE TITLE
Improve Survival Rate score

### ISSUE THREAD

**Issue by shinebrillant** [enrique-lozano/Monekin#248]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/enrique-lozano/Monekin/blob/main/CONTRIBUTING.md#reporting-issues).
- [X] I agree to follow the [Code of Conduct](https://github.com/enrique-lozano/Monekin/blob/main/CODE_OF_CONDUCT.md).
- [X] I have searched for [existing issues](https://github.com/enrique-lozano/Monekin/issues?q=is%3Aissue) that already report this problem, without success.
- [X] I have updated the app to the last released version.

### Current Behavior

The survival rate of more than 5 months and the score that can last a lifetime are both 50 points.

### Expected Behavior

Survival rate greater than 5 months can earn 49 points.
The survival rate can be maintained for a lifetime and you can get 50 points.

### Steps to Reproduce

Add a new income and go to the data management tool to view it.

### Additional Information

It is easy to maintain the survival rate for more than 5 months, but it is too difficult to maintain the survival rate for a lifetime, but there is almost no difference in the survival rate score.
There is no way to show that the survival rate score is greater than 5 months and less than a lifetime.

**Comment by shinebrillant** [comment]
All I can think of is 2 dividing 50 into two points, 25 points to score whether it is more than 5 months, and 25 points to score whether it can last a lifetime.

If you can maintain it for 5 months, you can get 25 points, if you can maintain it for a lifetime, you can get 24 points, and 1 point less is used as a reminder to continue working hard.

**Comment by enrique-lozano** [comment]
I think this score is more subjective than anything else. 

The truth is that I think that once you have the possibility of living 6 months with the same pace of life without any income, you should have a score very close to the maximum (50pts). This is the actual behaviour I think

**Comment by shinebrillant** [comment]
Can one more figure be added to calculate the cost of distance lasting a lifetime?
If you are willing, you can easily reach the cost of maintaining it for more than 5 months and get 50 points, but this is far from enough. I need an ugly score to alert myself, rather than using a good-looking score to deceive myself.

**Comment by enrique-lozano** [comment]
Yep, that make sense. Maybe we can modify a little bit the formula to calculate this score. Let's wait until someone have more ideas/suggestions

**Comment by RelapsingCertainly** [comment]
Just wanna throw my 2 cents in. 
I don't think a lifetime survival rating is entirely possible due to their being a lot of unpredictable things in life. Um, one way to improve the financial health score could be have a checklist of sorts where you can have a few requirements to get a 90+ Financial health score.
Some examples could be:
-  have a 6 - 12 month surplus in a savings account
- have a low or zero debt ratio (could look at accounts and check the ratio between the negative and positive balances)
- How close do someone follow their budget? (Add all budget and average usage percentage)

Honestly, is someone does meet all there requirements ( having a 12 month surplus, have a > 0.1 debt ratio and follows >= 100% of their set budgets), they are in a really really good position and they should be able dial back on saving and be able to spend. (Want to avoid a Scrooge situation lol)

**Comment by enrique-lozano** [comment]
> I now think that the survival rate (12 months) and savings ratio (20%) need to be reduced by 1 point respectively. Only more stringent conditions, such as requiring double, the survival rate needs to be 24 months, and the savings ratio needs to be 40 %, so that you can get the final 2 points.

Definitely yes, I think we can implement this for now. As always, I accept PRs on this matter 😊

> have a low or zero debt ratio (could look at accounts and check the ratio between the negative and positive balances)

I really like this idea and I definitely want to add more parameters that are used to calculate your financial health. However, I disagree with the way debts are calculated, as I think a negative balance should not necessarily be considered as a debt. 

I am thinking of perhaps implementing other types of accounts and having these be used to calculate financial health. For example, apart from the two that are already there, we could have debt accounts, investments... 

Regarding debt accounts, I think we should not take them into account when calculating your final balance, unlike what happens with the other types of accounts. That is, if the user takes out a mortgage of 200k, he could create a debt account with a balance of 200k and add the transactions of his mortgage to this account. When showing the final balance I don't think it would be good to show -200,000€ for example, instead we should ignore this account.

**Comment by RelapsingCertainly** [comment]
> However, I disagree with the way debts are calculated, as I think a negative balance should not necessarily be considered as a debt.

I also don't like negative balance being considered a debt in the long term, but it would be the fastest way to implement right now.

> When showing the final balance I don't think it would be good to show -200,000€ for example, instead we should ignore this account

This kinda links to my other discussion post about account visibility so there could be future update where the cashew system is copied for long term loans

**Comment by enrique-lozano** [comment]
I added a much more complex calculation for the financial score that will be out with Monekin version 10! You could find more details in the implementation PR: #506 

For further improvements in this regard, please create a more specific enhacement to this calculation system or just a discussion thread 🚀
